packages/twenty-front/src/modules/object-record/relation-picker/components/SingleEntitySelectMenuItemsWithSearch.tsx

```diff
@@ -36,11 +39,24 @@ export const SingleEntitySelectMenuItemsWithSearch = ({
   relationPickerScopeId = 'relation-picker',
   selectedEntity,
   selectedRelationRecordIds,
+  clearOnOpen = false,
 }: SingleEntitySelectMenuItemsWithSearchProps) => {
-  const { searchFilter, searchQuery, handleSearchFilterChange } =
-    useEntitySelectSearch({
-      relationPickerScopeId,
-    });
+  const {
+    searchFilter,
+    searchQuery,
+    handleSearchFilterChange,
+    resetSearchFilterChange,
+  } = useEntitySelectSearch({
+    relationPickerScopeId,
+  });
+
+  useEffect(() => {
```

Comment:
We avoid this pattern:
1) instead of using a useEffect, we prefer perform the action (here reset the search filter) sequentially when needed. For example, on onSubmit or on onCancel. This avoids creating side effects that are difficult to debug and also creating re-renders
2) here you are hacking the useEffect dependency array to avoid re-renders which is a symptom of the problem above

Reply:
hey @charlesBochet, Thanks for the Review, I've made changes so that we will reset the field when `onCancel` is triggered or an entity is selected